lib/spack/spack/directives.py

```diff
@@ -218,6 +218,8 @@ def _execute(pkg):
         if checksum:
             kwargs['md5'] = checksum
 
+        kwargs['version'] = ver
```

Comment:
Good detective work!  Can you please add the word `FetchStrategy` to the comments on that function, explaining this?  That way when someone greps for `FetchStrategy` they will find it.